sql/wsrep_mysqld.cc

```diff
@@ -2412,8 +2446,15 @@ void wsrep_to_isolation_end(THD *thd)
     wsrep_cleanup_transaction(thd);
   }
 
-  if (thd->wsrep_skip_wsrep_hton)
+  if (thd->wsrep_skip_wsrep_hton) {
+    DEBUG_SYNC(thd, "wsrep_to_isolation_end_before");
+    mysql_mutex_lock(&thd->LOCK_thd_data);
```

Comment:
Should't this be `thd->LOCK_wsrep_thd`?

Reply:
This is to match up with kill_one_thread() which uses LOCK_thd_data.